src/vcf/fixer.cpp

```diff
@@ -210,10 +238,149 @@ namespace ebi
+                util::string_split(*first, ":", format_keys);
+
+                for (size_t i = 0; i < format_keys.size(); i++) {
+                    format_fields_indexes[format_keys[i]].push_back(i);
+                    if (format_fields_indexes[format_keys[i]].size() == 1) {
+                        ordered_fields.push_back(format_keys[i]);
+                    }
+                }
+
+                std::set<std::string> fields_to_remove = get_format_fields_to_remove(format_fields_indexes, first, last);
+
+                std::vector<std::string> fixed_format;
+                for (auto & ordered_field : ordered_fields) {
+                    if (fields_to_remove.find(ordered_field) == fields_to_remove.end()) {
+                        // keep first occurrence, discard the rest if present
+                        fixed_format.push_back(ordered_field);
+                    } else {
+                        num_removed_duplicates++;
+                    }
+                }
+
+                if (fixed_format.empty()) {
+                    throw std::runtime_error("Could not fix FORMAT duplicate fields: All fields had to be removed, but missing value is not permitted");
+                }
+                util::print_container(output, fixed_format, "", ":","");
+
+                for (auto it = first + 1; it != last; it++) {
+                    std::vector<std::string> sample_fields;
+                    util::string_split(*it, ":", sample_fields);
+                    std::vector<std::string> fixed_sample;
+                    for (auto & ordered_field : ordered_fields) {
+                        if (fields_to_remove.find(ordered_field) == fields_to_remove.end()) {
+                            // keep first occurrence, discard the rest if present
+                            if (sample_fields.size() > format_fields_indexes[ordered_field][0]) {
+                                fixed_sample.push_back(sample_fields[format_fields_indexes[ordered_field][0]]);
+                            }
+                        }
+                    }
+                    output << "\t";
+                    if (fixed_sample.empty()) {
+                        output << MISSING_VALUE;         // all the sample fields were removed, so write missing value to output
+                    } else {
+                        util::print_container(output, fixed_sample, "", ":", "");
+                    }
+                }
+            });
+
+            return num_removed_duplicates;
+        }
+
+        std::set<std::string> Fixer::get_format_fields_to_remove(std::map<std::string, std::vector<size_t>> &format_fields_indexes,
+                                                                 std::vector<std::string>::iterator first,
+                                                                 std::vector<std::string>::iterator last)
+        {
+            std::set<std::string> fields_to_remove;
+
+            for (auto it = first + 1; it != last; it++) {
```

Comment:
Wouldn't it be more intuitive that `first + 1` was the value of the input parameter, instead of skipping the first element without making it explicit to external callers? If I was to call this method I would expect it to process all the elements from first to last, not to skip the first one.